keps/sig-auth/4317-pod-certificates/README.md

```diff
@@ -0,0 +1,1045 @@
+
+**Make it easy to ship secure third-party signer implementations**:  Securely implementing and operating a third-party signer implementation today requires intimate familiarity with Kubernetes' security model, otherwise your signer may negate Kubernetes' node restriction security boundary.  This KEP moves responsibilty for enforcing node restriction on CSRs into kube-apiserver, so that signer implementations do not need to consider it.
+
+### Non-Goals
+
+**Specify a solution for pod-to-pod mTLS in core Kubernetes**: Pod-to-pod mTLS (aka service mesh) is a rapidly-evolving area.  While the PodCertificate volume and PodIdentity X.509 extension are expected (and designed to be) useful for service mesh implementations, it doesn't make sense to try and standardize pod-to-pod mTLS at the core Kubernetes layer today.
+
+**Support TPM-backed private keys**: In this proposal, private keys are
+contained to the node (held only in kubelet memory and tmpfs projected volumes),
+but they can still be extracted from a compromised workload (or compromised
+node).  Storing the private keys in TPMs closes this loophole, but will require
+an additional KEP to define how kubelet can use the Linux (or other platform's)
+virtual TPM subsystem to provide virtual TPMs to pods
+
+**Be 100% compatible with software in the field today**: The primary example in this KEP is that the private key and certificate chain are delivered in a single credential-bundle file.  This is based on implementation and operation experience of a CSI-based certificates solution that delivered private keys and certificate chains in separate files, as expected by most commodity webservers today.  While this is an attractive idea, automatically rotating these separate files is fraught.  Even if both file's content is updated atomically, skewed reads in the workload can result in hard-to-diagnose errors if the workload reads the key before rotation and the certificate chain after rotation (or vice-versa).
+
+**Support private keys shared across multiple pods**: Sharing private keys and certificates across multiple pods may be required in certain applications.  However, this use case is well-served by Secret projected volume sources.
+
+**Support human-in-the-loop approval for certificate requests**: This KEP assumes that CSR approval and issuance is fully-automated, because this process blocks pod startup for pods that use PodCertificate projected volume sourcs.
+
```

Comment:
Probable non-goals to add, that have cropped up as concerns in parallel impls (e.g. SPIRE) that are by-definition out of scope for this as written:

* Supporting identity delegation (minting certs that can be used by workloads for intermediate CA purposes)
* Supporting identity impersonation (one privileged workload minting many certs on-behalf-of other workloads)

Reply:
Captured as an unresolved block in the non-goals section.